### PR TITLE
[DDCE-3505] Mandated PAYE Reference + add new unit tests

### PR DESCRIPTION
**Pull request by matthewwardcap**
# DDCE-3505

- Dependencies Updated
- Some missing tests were added back in
- Mandated paye reference fields
- Updated unit tests

# PLEASE READ BEFORE REVIEW:
This change has to be left un-merged as part of the CR, this will be merged later sometime in April.

Acceptance tests will need to be compared against the DDCE-3505 branch of `ers-checking-acceptance-tests` as it also can't be merged until April (feature switch)

## Checklist PR Raiser
 - [X]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [X]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [X]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [X]  I've squashed my commits - including the JIRA issue number in the commit message
 - [X]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: conf/validation-config/EMI/ers-emi-taxable-validation.conf [hunk 1]
```diff
@@ -101,7 +101,7 @@ ers-emi-taxable-validation-config {
     {
       column = "H"
       cellName = "8. PAYE reference"
-      mandatory = false
+      mandatory = true
       error: {
           id = "error.8"
           errorMsg = ${validation-types.paye.errorMsg}
```
Comment (slavvco) [inline conf/validation-config/EMI/ers-emi-taxable-validation.conf:107]: Just to confirm, here the error msg/rule need no updating?

File: conf/validation-config/SAYE/ers-saye-rcl-validation.conf [hunk 2]
```diff
@@ -101,7 +101,7 @@ ers-saye-rcl-validation-config {
     {
       column = "H"
       cellName = "8. PAYE reference"
-      mandatory = false
+      mandatory = true
       error: {
         id = "error.8"
         errorMsg = ${validation-types.paye.errorMsg}
```
Comment (slavvco) [inline conf/validation-config/SAYE/ers-saye-rcl-validation.conf:107]: Just to confirm, here the error msg/rule need no updating?

File: conf/validation-config/EMI/ers-emi-taxable-validation.conf (hunk 1 above)
Comment (matthewwardcap) [reply conf/validation-config/EMI/ers-emi-taxable-validation.conf:107]: Yes this is correct. I double checked this with @DannyPennington97 and Muhammed and due to the fact this PAYE reference isn't specific to emp like the rest of the changes, can just be left the same for now

File: conf/validation-config/SAYE/ers-saye-rcl-validation.conf (hunk 2 above)
Comment (matthewwardcap) [reply conf/validation-config/SAYE/ers-saye-rcl-validation.conf:107]: Yes this is correct. I double checked this with @DannyPennington97 and Muhammed and due to the fact this PAYE reference isn't specific to emp like the rest of the changes, can just be left the same for now

**Review by slavvco — APPROVED** [review]

**Pull request closed without merging**